### PR TITLE
Recursor: replace/remove jQuery

### PR DESCRIPTION
**Pull request by zeha**
### Short description

Everything we used from jQuery is now available without hacks/quirks in plain JS. Yay 2020+.

Saves updating jQuery, too.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by zeha** [comment]
I have some time this week. If you can stomach the review, I could also:
1) do some more tidying up in the recursor html/js stuff
2) then copy/sync to dnsdist

**Comment by omoerbeek** [comment]
I'm not the right person to review the diff, I do not speak javascript. But I can test and will do that.

**Comment by omoerbeek** [comment]
Tested with Chrome and Safari and things look ok, but Firefox does not work:
JS console shows:
```
Uncaught ReferenceError: $ is not defined
    <anonymous> http://127.0.0.1:8083/local.js:6
```

**Comment by zeha** [comment]
```
Uncaught ReferenceError: $ is not defined
    <anonymous> http://127.0.0.1:8083/local.js:6
```
I did most of my dev/tests with Firefox. `local.js` no longer has a `$` on line 6. Can you clear your cache and see if that helps?

**Comment by zeha** [comment]
> I did most of my dev/tests with Firefox. `local.js` no longer has a `$` on line 6. Can you clear your cache and see if that helps?

if that helps, I'm inclined to rename local.js to a new filename to avoid the cache problem...

**Comment by Habbie** [comment]
> if that helps, I'm inclined to rename local.js to a new filename to avoid the cache problem...

Let's just do this, please :)

**Comment by kpfleming** [comment]
Sorry for the drive-by comment, but I believe those `script` tags can include a hash of the linked-to content, and that helps the browser know when it should throw away cached copies.

**Comment by zeha** [comment]
> Sorry for the drive-by comment, but I believe those `script` tags can include a hash of the linked-to content, and that helps the browser know when it should throw away cached copies.

Yeah, I think we don't have the appropriate tooling for that right now.

**Comment by zeha** [comment]
> > if that helps, I'm inclined to rename local.js to a new filename to avoid the cache problem...
> 
> Let's just do this, please :)

Pushed a commit for this.

**Comment by omoerbeek** [comment]
I get these errors now:
```
Aug 16 10:07:05 msg="Incomplete request" subsystem="webserver" level="0" prio="Debug" tid="0" ts="1660637225.019"
Aug 16 10:07:05 msg="Error result" error="Unknown error: 400" subsystem="webserver" level="0" prio="Debug" tid="0" ts="1660637225.019" urlpath=""
Aug 16 10:07:05 msg="Request" subsystem="webserver" level="0" prio="Info" tid="0" ts="1660637225.019" HTTPVersion="1.1" method="" remote="127.0.0.1:58980" respsize="183" status="400" uniqueid="95825e26-df0c-44f4-8b0d-190553af1dfe" urlpath=""
Aug 16 10:07:05 msg="Handling request" subsystem="webserver" level="0" prio="Debug" tid="0" ts="1660637225.021" uniqueid="bb2f3bbf-56ba-4119-bb76-46f5df0e5373" urlpath="/"
Aug 16 10:07:05 msg="Result" subsystem="webserver" level="0" prio="Debug" tid="0" ts="1660637225.021" bodyLength="5410" status="200" uniqueid="bb2f3bbf-56ba-4119-bb76-46f5df0e5373" urlpath="/"
Aug 16 10:07:05 msg="Request" subsystem="webserver" level="0" prio="Info" tid="0" ts="1660637225.021" HTTPVersion="1.1" method="GET" remote="127.0.0.1:58981" respsize="5734" status="200" uniqueid="bb2f3bbf-56ba-4119-bb76-46f5df0e5373" urlpath="/index.html"
Aug 16 10:07:05 msg="Handling request" subsystem="webserver" level="0" prio="Debug" tid="0" ts="1660637225.034" uniqueid="00017b49-394e-4d28-846d-06d391986df2" urlpath="/js/d3.v3.js"
Aug 16 10:07:05 msg="Result" subsystem="webserver" level="0" prio="Debug" tid="0" ts="1660637225.034" bodyLength="144718" status="200" uniqueid="00017b49-394e-4d28-846d-06d391986df2" urlpath="/js/d3.v3.js"
Aug 16 10:07:05 msg="Request" subsystem="webserver" level="0" prio="Info" tid="0" ts="1660637225.035" HTTPVersion="1.1" method="GET" remote="127.0.0.1:58982" respsize="145057" status="200" uniqueid="00017b49-394e-4d28-846d-06d391986df2" urlpath="/js/d3.v3.js"
```

File: pdns/recursordist/html/local-2022.js [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
+    headers: {'Accept': 'application/json', 'X-API-Key': 'changeme'},
+};
+*/
+
 var gdata = {};
 
-$(document).ready(function () {
-    $.ajaxSetup({cache: false});
+function get_json(url, params) {
+    const realURL = new URL(url + '?' + (new URLSearchParams(params)).toString(), fetchConfig.baseURL);
```
Comment (ahouene) [inline pdns/recursordist/html/local-2022.js:20]: This works in our usecases but could easily break, could you rather use `realURL.append` on each key/value pairs in `params` please?

Something such as:
```js
const realURL = new URL(url, fetchConfig.baseURL);
for (const [k, v] of Object.entries(params)) {
    realURL.append(k, v);
}
```

File: pdns/recursordist/html/local-2022.js [hunk 2]
```diff
@@ -1,13 +1,44 @@
...
 
-// var moment= require('moment');
+const fetchConfig = {
+    baseURL: window.location,
+    mode: 'same-origin',
+    headers: {'Accept': 'application/json'},
+};
+/*
+// Useful for development of the embedded webserver files.
+const fetchConfig = {
```
Comment (ahouene) [inline pdns/recursordist/html/local-2022.js:10]: This commented snippet can be confusing. Can you remove it and set `X-API-Key` by default?

**Review by ahouene — CHANGES_REQUESTED** [review]

File: pdns/recursordist/html/local-2022.js (hunk 2 above)
Comment (zeha) [reply pdns/recursordist/html/local-2022.js:10]: no, that will not work. in a production build, x-api-key should not be set - its only useful if the html/js files are used "unembedded".

Comment (ahouene) [reply pdns/recursordist/html/local-2022.js:10]: You're right - my bad!

File: pdns/recursordist/html/local-2022.js (hunk 1 above)
Comment (zeha) [reply pdns/recursordist/html/local-2022.js]: done!

**Review by ahouene — APPROVED** [review]

**Review by omoerbeek — APPROVED** [review]

**Pull request merged**